File: 백준/Silver/1931. 회의실 배정/회의실 배정.cc
#include <bits/stdc++.h>
using namespace std;

int n,last,first;
int period,cnt=1;
vector<pair<int,int>> clndr;

int main()
{
    cin>>n;
    for(int i=0;i<n;i++)
    {
        cin>>first>>last;
        clndr.push_back(make_pair(last,first));
    }
    sort(clndr.begin(),clndr.end());
    period=clndr[0].first;
    for(int i=1;i<n;i++)
    {
        if(period<=clndr[i].second)
        {
            cnt++;
            period=clndr[i].first;
        }
    }
    cout<<cnt;
    return 0;
}

File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
n = int(input())
arr = []
for _ in range(n):
    arr.append(list(map(int, input().split())))

arr.sort(key=lambda x: (x[1], x[0]))

cnt = 0
end = 0

for a in arr:
    if a[0] >= end:
        cnt += 1
        end = a[1]

print(cnt)
